File: src/test/java/pages/LoginPage.java
package pages;

import actions.BaseActions;
import org.openqa.selenium.By;

/**
 * Page Object for the myQ login screen.
 *
 * NOTE: accessibility ids below are placeholders. Inspect the real app
 * with Appium Inspector / Xcode Accessibility Inspector and update these
 * to match the actual accessibilityIdentifier (iOS) / content-desc
 * (Android) values before running against a real build.
 */
public class LoginPage extends BaseActions<LoginPage> {

    private final By emailInput = accessibilityId("email-input");
    private final By passwordInput = accessibilityId("password-input");
    private final By loginButton = accessibilityId("login-button");
    private final By errorMessage = accessibilityId("login-error-message");
    private final By forgotPasswordLink = accessibilityId("forgot-password-link");
    private final By dashboardScreen = accessibilityId("dashboard-screen");

    public LoginPage enterEmail(String email) {
        return type(emailInput, email);
    }

    public LoginPage enterPassword(String password) {
        return type(passwordInput, password);
    }

    public LoginPage tapLogin() {
        return tap(loginButton);
    }

    /** Full login flow, chainable: loginPage.login(email, password).isErrorDisplayed(); */
    public LoginPage login(String email, String password) {
        return enterEmail(email).enterPassword(password).tapLogin();
    }

    public String getErrorMessage() {
        return getText(errorMessage);
    }

    public boolean isErrorDisplayed() {
        return isDisplayed(errorMessage);
    }

    public boolean isLoginScreenDisplayed() {
        return isDisplayed(loginButton);
    }

    public boolean isDashboardDisplayed(int timeoutSeconds) {
        return isDisplayed(dashboardScreen, timeoutSeconds);
    }

    public LoginPage tapForgotPassword() {
        return tap(forgotPasswordLink);
    }
}

File: pages/LoginPage.js
const BasePage = require('./BasePage');

/**
 * Page Object for the myQ login screen.
 *
 * NOTE: accessibility ids below are placeholders. Inspect the real app
 * with Appium Inspector / Xcode Accessibility Inspector and update these
 * to match the actual `accessibilityIdentifier` (iOS) / `content-desc`
 * (Android) values before running against a real build.
 */
class LoginPage extends BasePage {
  constructor(driver) {
    super(driver);

    this.selectors = {
      emailInput: this.accessibilityId('email-input'),
      passwordInput: this.accessibilityId('password-input'),
      loginButton: this.accessibilityId('login-button'),
      errorMessage: this.accessibilityId('login-error-message'),
      forgotPasswordLink: this.accessibilityId('forgot-password-link'),
      dashboardScreen: this.accessibilityId('dashboard-screen'),
    };
  }

  async enterEmail(email) {
    await this.setValue(this.selectors.emailInput, email);
  }

  async enterPassword(password) {
    await this.setValue(this.selectors.passwordInput, password);
  }

  async tapLogin() {
    await this.tap(this.selectors.loginButton);
  }

  /**
   * Full login flow: fill credentials and submit.
   */
  async login(email, password) {
    await this.enterEmail(email);
    await this.enterPassword(password);
    await this.tapLogin();
  }

  async getErrorMessage() {
    return this.getText(this.selectors.errorMessage);
  }

  async isErrorDisplayed() {
    return this.isDisplayed(this.selectors.errorMessage);
  }

  async isLoginScreenDisplayed() {
    return this.isDisplayed(this.selectors.loginButton);
  }

  async waitForDashboard(timeout) {
    return this.waitForDisplayed(this.selectors.dashboardScreen, timeout);
  }
}

module.exports = LoginPage;
